### PR TITLE
hieroglyphic: init at 1.0.1

### LINKED ISSUE #307577: Package request: Hieroglyphic
**Project description**

Local only version of detexify. Drawings to LaTeX command converter.

**Metadata**

* homepage URL: https://flathub.org/apps/io.github.finefindus.Hieroglyphic
* source URL: https://github.com/FineFindus/Hieroglyphic
* license: GPL 3.0
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/307577

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/hi/hieroglyphic/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hieroglyphic";
+  version = "1.0.1";
+  # Note: 1.1.0 requires a higher gtk4 version. This requirement could be patched out.
+
+  src = fetchFromGitHub {
+    owner = "FineFindus";
+    repo = "Hieroglyphic";
+    rev = "v${finalAttrs.version}";
```
Comment (eclairevoyant) [inline pkgs/by-name/hi/hieroglyphic/package.nix:26]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

File: pkgs/by-name/hi/hieroglyphic/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  ];
+
+  meta = {
+    changelog = "https://github.com/FineFindus/Hieroglyphic/releases/tag/${finalAttrs.src.rev}";
+    description = "A tool based on detexify for finding LaTeX symbols from drawings";
+    homepage = "https://apps.gnome.org/en/Hieroglyphic/";
+    license = lib.licenses.gpl3Only;
+    mainProgram = "hieroglyphic";
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platfomrs = lib.platfomrs.linux ++ lib.platfomrs.darwin;
```
Comment (eclairevoyant) [inline pkgs/by-name/hi/hieroglyphic/package.nix:60]:
```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

File: pkgs/by-name/hi/hieroglyphic/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  ];
+
+  buildInputs = [
+    glib
+    gtk4
+    libadwaita
+  ];
+
+  meta = {
+    changelog = "https://github.com/FineFindus/Hieroglyphic/releases/tag/${finalAttrs.src.rev}";
```
Comment (eclairevoyant) [inline pkgs/by-name/hi/hieroglyphic/package.nix:54]:
```suggestion
    changelog = "https://github.com/FineFindus/Hieroglyphic/releases/tag/v${finalAttrs.version}";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/by-name/hi/hieroglyphic/package.nix (hunk 2 above)
Comment (TomaSajt) [reply pkgs/by-name/hi/hieroglyphic/package.nix]: Fixed

File: pkgs/by-name/hi/hieroglyphic/package.nix (hunk 1 above)
Comment (TomaSajt) [reply pkgs/by-name/hi/hieroglyphic/package.nix]: Done

File: pkgs/by-name/hi/hieroglyphic/package.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  ];
+
+  meta = {
+    changelog = "https://github.com/FineFindus/Hieroglyphic/releases/tag/v${finalAttrs.version}";
+    description = "A tool based on detexify for finding LaTeX symbols from drawings";
+    homepage = "https://apps.gnome.org/en/Hieroglyphic/";
+    license = lib.licenses.gpl3Only;
+    mainProgram = "hieroglyphic";
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platfomrs = lib.platforms.linux ++ lib.platforms.darwin;
```
Comment (eclairevoyant) [inline pkgs/by-name/hi/hieroglyphic/package.nix:60]:
```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

**Review by eclairevoyant — DISMISSED** [review]

File: pkgs/by-name/hi/hieroglyphic/package.nix (hunk 4 above)
Comment (eclairevoyant) [reply pkgs/by-name/hi/hieroglyphic/package.nix:60]: btw the meta checks can be run locally to catch these without having to wait for borgo:
https://github.com/NixOS/ofborg?tab=readme-ov-file#running-meta-checks-locally

Comment (TomaSajt) [reply pkgs/by-name/hi/hieroglyphic/package.nix:60]: Good to know, thanks for the info, I'll check it out

Comment (TomaSajt) [reply pkgs/by-name/hi/hieroglyphic/package.nix]: Fixed

Comment (TomaSajt) [reply pkgs/by-name/hi/hieroglyphic/package.nix]: Oh, committed too much

**Review by eclairevoyant — APPROVED** [review]

**Pull request merged**